benches/bench.rs

```diff
@@ -170,3 +170,118 @@ fn size_22_4m(b: &mut test::Bencher) {
+        while i < to_send {
+            let (tx, rx) = ipc::channel().unwrap();
+            rx_set.add(rx).unwrap();
+            active.push(tx);
+            i += 1;
+        }
+        while i < n {
+            let (tx, rx) = ipc::channel::<()>().unwrap();
+            rx_set.add(rx).unwrap();
+            dormant.push(tx);
+            i += 1;
+        }
+        b.iter(|| {
+            for tx in active.iter() {
+                tx.send(()).unwrap();
+            }
+            let mut received = 0;
+            while received < to_send {
+                for result in rx_set.select().unwrap().into_iter() {
+                    let (_, _) = result.unwrap();
+                    received += 1;
+                }
+            }
+        });
+    }
+
+    fn create_empty_set() -> Result<IpcReceiverSet, ()> {
+        Ok(IpcReceiverSet::new().unwrap())
+    }
+
+    fn add_n_rxs(rx_set: &mut IpcReceiverSet, n: usize) -> () {
+        for _ in 0..n {
+            let (_, rx) = ipc::channel::<()>().unwrap();
+            rx_set.add(rx).unwrap();
+        }
+    }
+
+    #[bench]
+    fn bench_one_percent_100(b: &mut test::Bencher) -> () {
+        gen_select_test(b, 1, 100);
+    }
+
+    #[bench]
+    fn bench_one_hundred_percent_100(b: &mut test::Bencher) -> () {
+        gen_select_test(b, 100, 100);
+    }
+
+    #[bench]
+    fn bench_twenty_percent_10(b: &mut test::Bencher) -> () {
+        gen_select_test(b, 2, 10);
+    }
+
+    #[bench]
+    fn bench_thirty_percent_10(b: &mut test::Bencher) -> () {
+        gen_select_test(b, 3, 10);
+    }
+
+    #[bench]
+    fn bench_twenty_percent_5(b: &mut test::Bencher) -> () {
+        gen_select_test(b, 1, 5);
+    }
+
+    #[bench]
+    fn bench_create_and_destroy_empty(b: &mut test::Bencher) -> () {
+        b.iter(|| {
+            create_empty_set().unwrap();
+        });
+    }
+
+    #[bench]
+    fn bench_create_and_destroy_10(b: &mut test::Bencher) -> () {
+        b.iter(|| {
+            let mut rx_set = IpcReceiverSet::new().unwrap();
+            add_n_rxs(&mut rx_set, 10);
+        });
+    }
+
+    #[bench]
+    fn bench_create_and_destroy_5(b: &mut test::Bencher) -> () {
+        b.iter(|| {
+            let mut rx_set = IpcReceiverSet::new().unwrap();
+            add_n_rxs(&mut rx_set, 5);
+        });
+    }
+
+    #[bench]
+    // Benchmark adding and removing closed receivers from the set
+    fn bench_manipulation(b: &mut test::Bencher) -> () {
+        b.iter(|| {
+            let mut rx_set = IpcReceiverSet::new().unwrap();
+            {
+                {
+                    let (_, rx) = ipc::channel::<()>().unwrap();
+                    rx_set.add(rx).unwrap();
+                }
+                // On select Receivers with a "ClosedChannel" event
+                // will be closed
+                rx_set.select().unwrap();
+                let (tx, rx) = ipc::channel::<()>().unwrap();
+                rx_set.add(rx).unwrap();
+                tx.send(()).unwrap();
```

Comment:
I'm still confused: doesn't adding another channel and doing a send just blur the results of the channel close handling, which is what we actually want to measure here; while the other operations should already be covered by the other tests?... Or is it somehow needed for semantic reasons? Am I missing something here?...